api/cpp/CMakeLists.txt

```diff
@@ -373,11 +373,30 @@ if (SLINT_BUILD_RUNTIME)
         include(${CMAKE_CURRENT_LIST_DIR}/cmake/SlintMacro.cmake)
     endif()
 
-    export(TARGETS Slint slint_cpp
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm)$")
+        add_library(STM32Platform INTERFACE)
+        target_sources(STM32Platform INTERFACE
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/stm32/slint-stm.cpp>
+            $<INSTALL_INTERFACE:include/slint/stm32/slint-stm.cpp>
+        )
+        target_include_directories(STM32Platform INTERFACE
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/stm32>
+            $<INSTALL_INTERFACE:include/slint/stm32>
+        )
+        target_link_libraries(STM32Platform INTERFACE Slint::Slint)
+
+        add_library(Slint::STM32Platform ALIAS STM32Platform)
+        set(stm32_platform_target STM32Platform)
+    endif()
+
+    export(TARGETS Slint slint_cpp ${stm32_platform_target}
         NAMESPACE Slint:: FILE "${CMAKE_BINARY_DIR}/lib/cmake/Slint/SlintTargets.cmake")
     install(EXPORT SlintTargets NAMESPACE Slint:: DESTINATION lib/cmake/Slint)
-    install(TARGETS Slint slint_cpp
+    install(TARGETS Slint slint_cpp ${stm32_platform_target}
         EXPORT SlintTargets LIBRARY DESTINATION lib PUBLIC_HEADER DESTINATION include/slint)
+    if (TARGET Slint::STM32Platform)
+        install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/stm32/slint-stm.cpp ${CMAKE_CURRENT_SOURCE_DIR}/stm32/slint-stm.h DESTINATION include/slint/stm32)
```

Comment:
Strange to have a .cpp file in include. But I guess it can't hurt.
Although do we actually need a .cpp file? Can we not just have a header-only code and put everything in the .h as inline?

Reply:
I'd really like to avoid a header file with things like this: https://github.com/slint-ui/slint/blob/36e0328cf34523a1be4dc7395204eb00ca3dc623/api/cpp/stm32/slint-stm.cpp#L11

On the other hand I think a cpp file is clean and there's no risk that anything we declare in the header affects the application build (especially implicit includes or macros from the BSP files we include).

But I admit that at this point my desperation to get this merged outweighs pretty much everything else. I can try tomorrow to get it to work as a header only implementation.

Reply:
One upside I can see of a header only solution is that it simplifies support for other build systems.

Reply:
> there's no risk that anything we declare in the header affects the application build (especially implicit includes or macros from the BSP files we include).

This would be a valid point if the STRINGIFY/CAT macro and bsp #include were in the .cpp file. But right now they are in the header.

Not sure what `__IO` means, but it could be a `inline static __IO bool StmWindowAdapter::screen_ready = true;`

Reply:
You're right. A few hours of sleep over it and I'm convinced about the header-only approach :)